src/calendar.js

```diff
@@ -480,16 +496,30 @@ export default class Calendar extends Emitter {
   }
 
   highLightRange(start, end) {
-    let invalidRangeMessage;
     const { range, isValid } = this.selectRange(start, end);
-    const minStay            = this.opts.rentalId ? (this.opts.allowShorterMinStaySelection ? 1 : this.cTree.getDayProperty(...start, 'minStay')) : this.opts.minStay;
-    let maxStay              = this.opts.rentalId ? (this.opts.allowLongerMaxStaySelection ? 1 : this.cTree.getDayProperty(...start, 'maxStay')) : this.opts.maxStay;
-
-    maxStay = (maxStay === 0 || !maxStay) ? Infinity : maxStay;
+    const { opts }           = this;
+    let { minStay, maxStay } = opts;
+    let invalidRangeMessage;
 
     if (isArray(range)) {
+      if (opts.rentalId) {
+        if (!opts.allowShorterMinStaySelection) {
+          // get the biggest minStay value in the range
+          minStay = Math.max(...range.map(function(cell) {
+            return cell.dataset.minStay || minStay;
+          }));
+        }
+
+        if (!opts.allowLongerMaxStaySelection) {
+          // get the biggest maxStay value in the range
+          maxStay = Math.max(...range.map(function(cell) {
+            return cell.dataset.maxStay || maxStay;
+          }));
+        }
```

Comment:
AFAICT this means that there is the possibility to have number of minimum / maximum nights defined for each day, and not just globally? That would introduce some interesting problems.

Reply:
@abel-n that's a good question, by doing this you could have a global minStay of 5 but one period from bookings data that would have a minStay of 10. But if you allow allowShorterMinStaySelection that would use the global minStay so I guess it's more flexible. ````